src/EditorConfigFile.php

```diff
@@ -145,4 +146,24 @@ private function parseIniString(string $content) : array
 
         return is_array($parsedContent) === true ? $parsedContent : [];
     }
+
+    private function cleanContent(string $path) : string
+    {
+        $handle = fopen($path, 'r');
+        $content = '';
+        if ($handle) {
+            while (($line = fgets($handle)) !== false) {
+                # We want to prevent this error: https://github.com/idiosyncratic-code/editorconfig-php/issues/3
+                if (preg_match('~#\[~', $line) === 1) {
+                    continue;
+                }
+
+                $content .= $line;
+            }
+
+            fclose($handle);
+        }
+
+        return $content;
+    }
```

Comment:
According to the EditorConfig spec:

> Comment: starts with a ; or a #.
> Inserting a # or ; after non-whitespace characters in a line (i.e., inline) shall neither be parsed as a comment nor as part of > the section name, pair (defined below) key or value in which it was inserted. This may change in the future; thus, is not recommended.

While this works to solve lines that start with a `#`, it will also remove lines with a trailing comment (which is technically not a comment at all according to the spec, but still).

I think it would be more correct to strip everything from `#` to the end of the line, something like:

```
    private function cleanContent(string $path) : string
    {
        $content = file_get_contents($path);

        $content = preg_replace('/#.*$/m', '', $content) ?? $content;

        return $content;
    }
```

Thoughts?

Reply:
Thanks for elaborating. Looks good to me and works perfectly.